### PR TITLE
Remove TitleLegendHeader

### PR DESCRIPTION
**Pull request by ranquild**
Refactoring to make vertical legends easier. Replaces `TitleLegendHeader` with a separate component designed to be used as a header. The old component was a wrapper around legend and passed custom `series` to make it render like a header.

### CODE REVIEW

File: frontend/src/metabase/visualizations/components/Visualization.jsx [hunk 1]
```diff
@@ -519,15 +519,13 @@ export default class Visualization extends React.PureComponent {
         className={cx(className, "flex flex-column full-height")}
         style={style}
       >
-        {!!hasLegendHeader && (
+        {!!hasHeader && (
           <div className="p1 flex-no-shrink">
-            <TitleLegendHeader
-              classNameWidgets={classNameWidgets}
```
Comment (ranquild) [inline frontend/src/metabase/visualizations/components/Visualization.jsx:525]: You can check `TitleLegendHeader`, this prop was never used

File: frontend/src/metabase/visualizations/components/Visualization.jsx [hunk 2]
```diff
@@ -519,15 +519,13 @@ export default class Visualization extends React.PureComponent {
...
+            <ChartHeader
               series={series}
-              actionButtons={extra}
-              description={settings["card.description"]}
```
Comment (ranquild) [inline frontend/src/metabase/visualizations/components/Visualization.jsx:528]: Same as this one.

File: frontend/src/metabase/visualizations/components/ChartHeader.jsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+const propTypes = {
+  series: PropTypes.array.isRequired,
+  settings: PropTypes.object.isRequired,
+  icon: PropTypes.shape(iconPropTypes),
+  actionButtons: PropTypes.node,
+  onChangeCardAndRun: PropTypes.func,
+};
+
+const ChartHeader = ({
```
Comment (ranquild) [inline frontend/src/metabase/visualizations/components/ChartHeader.jsx:14]: Ideally this should be a hook, but we are going to replace the component used in classes only, so I made it a component as well.

**Review by alxnddr — COMMENTED** [review]
Looks good and more readable 👍 

I noticed one tiny visual regression:

Before
Screen Shot 2021-08-10 at 17 08 27

After
Screen Shot 2021-08-10 at 17 33 34


You can find this screen by enabling sharing and embedding in admin panel and then:
Screen Shot 2021-08-10 at 17 35 01

Screen Shot 2021-08-10 at 17 35 06

File: frontend/src/metabase/visualizations/components/ChartCaption.jsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+};
+
+const ChartCaption = ({
+  series,
+  settings,
+  icon,
+  actionButtons,
+  onChangeCardAndRun,
+}) => {
+  const title = settings["card.title"] || series[0].card.name;
```
[the remark this answers is not in the document]
Comment (ranquild) [reply frontend/src/metabase/visualizations/components/ChartCaption.jsx]: Changed the code as you suggested. Could you please check that the issue is solved? I wasn't able to reproduce it with Pie charts and dashboard embedding.

**Review by alxnddr — APPROVED** [review]

**Pull request merged**